src/utils/Utils.ts

```diff
@@ -21,11 +21,15 @@ export class Utils {
         );
       })
       .filter((record: any) => {
-        // Filter by testTypeClassification
+        // Filter by testTypeClassification or testTypeClassification, testResult and ivaDefects present and populated
         if (record.testTypes && record.testTypes.testTypeClassification) {
           return (
-            record.testTypes.testTypeClassification ===
-            "Annual With Certificate"
+            record.testTypes.testTypeClassification === "Annual With Certificate" ||
+            (
+                record.testTypes.testTypeClassification === "IVA With Certificate" &&
+                record.testTypes.testResult === "fail" &&
+                record.testTypes.ivaDefects && record.testTypes.ivaDefects.length > 0
+            )
           );
         }
```

Comment:
I personally find this starting to be difficult to understand what's happening with the chaining, maybe we could do something like this. What's your thoughts?
```
        .filter((record: any) => {
          const testTypes = record.testTypes;

          const isAnnualWithCertificate = testTypes?.testTypeClassification === "Annual With Certificate";
          const isIvaWithCertificate = testTypes?.testTypeClassification === "IVA With Certificate";
          const isTestResultFail = testTypes?.testResult === "fail";
          const hasNonEmptyIvaDefects = !!testTypes?.ivaDefects?.[0];

          return isAnnualWithCertificate || (isIVAWithCertificate && isTestResultFail && hasNonEmptyIvaDefects);
        });
```

Reply:
Happy if we want to separate the elements out in this fashion, although the bottom `const` I would prefer a length check i.e.

`const hasNonEmptyIvaDefects = !!(testTypes?.ivaDefects?.length);`

If the length is 0, then it will return false, and if it's greater than 0 then true.

Reply:
I like this, it does make it more readable now there are more conditions

Reply:
I'm not precious either so happy to change to make more readable.